system-addon/common/Reducers.jsm

```diff
@@ -84,7 +84,7 @@ function insertPinned(links, pinned) {
   // Then insert them in their specified location
   pinned.forEach((val, index) => {
     if (!val) { return; }
-    let link = Object.assign({}, val, {isPinned: true, pinIndex: index});
+    let link = Object.assign(val, {isPinned: true, pinIndex: index});
```

Comment:
What's your reasoning behind this change? Mutating stuff that comes from function arguments seems like a bad idea to me

Reply:
This is so that we keep the original pinned link object. Otherwise screenshots are saved to this copy instead of to the one in the cache.

Reply:
Restored the `.assign({}`